### PR TITLE
Dashboard - new testing framework: playwright

### PR DESCRIPTION
**Pull request by notoraptor**
# Description

Hi @bouthilx ! This is a PR that adds all changes related to unit tests, excluding changes related to experiment progress bar.

As this work was done to be able to test infinite scroll, I kept infinite scrolling feature added in this PR.

This PR replaces old dashboard testing framework [React testing library](https://testing-library.com/docs/react-testing-library/api/) with a new framework [Playwright](https://github.com/microsoft/playwright) which uses a real browser in backend, allowing to make more complex tests, especially tests that needs HTML components with real dimensions. This is particularly necessary to test how a component reacts to a scrolling.

# Changes

- Add infinite scrolling to experiments nav bar
- Replace React Testing Library with Playwright
- Rewrite a new Github Actions to use Playwright
- Rewrite all old tests to work with Playwright
- Add a new test for nav bar scrolling in `dashboard/src/src/__tests__/ExperimentNavBar.test.js`
- Add a new PKL database containing supplementary uncompleted experiments. These experiments are not yet used here, as it is mainly to test progress bars, but I leave it here as it changes a single line in tests. I can remove it from this PR if necessary. Other related changes:
  - Add a script to add supplementary experiments, to use if we need to regenerate the new DB
  - Update the script that copy data from PKL db to Mongo db, to use new PKL db as source, and clean Mongo DB before copying new data. This script is used to populate a Mongo DB in CI.

# Checklist

## Tests
- [X] I added corresponding tests for bug fixes and new features. If possible, the tests fail without the changes
- [ ] All new and existing tests are passing (`$ tox -e py38`; replace `38` by your Python version if necessary)

## Documentation
- [ ] I have updated the relevant documentation related to my changes

## Quality
- [ ] I have read the [CONTRIBUTING](https://github.com/Epistimio/orion/blob/develop/CONTRIBUTING.md) doc
- [ ] My commits messages follow [this format](https://chris.beams.io/posts/git-commit/)
- [ ] My code follows the style guidelines (`$ tox -e lint`)

### CODE REVIEW

File: dashboard/src/src/__tests__/VisualizationsPage.test.js [hunk 1]
```diff
@@ -1,135 +1,115 @@
...
+  test('Test if we can select and unselect experiments', async ({ page }) => {
+    const firstExperiment = await page.getByText(/2-dim-shape-exp/);
+    await firstExperiment.waitFor();
 
-  // Switch to visualizations page
-  const menu = screen.queryByTitle(/Go to experiments visualizations/);
-  fireEvent.click(menu);
-  expect((await screen.findAllByText(/Nothing to display/)).length).toBe(3);
+    // Go to visualization page
+    const menuExperiments = await page.locator('nav > ul > li:nth-child(1)');
```
Comment (bouthilx) [inline dashboard/src/src/__tests__/VisualizationsPage.test.js:46]: Are the elements more difficult to find using playwright?

**Review by bouthilx — APPROVED** [review]
LGTM, thanks! I relaunched the tests. I don't understand how the current failures would relate to the current changes in this PR.

File: dashboard/src/src/__tests__/VisualizationsPage.test.js (hunk 1 above)
Comment (notoraptor) [reply dashboard/src/src/__tests__/VisualizationsPage.test.js:46]: No, it is pretty simple ! It was more difficult with puppeteer, and that's why I searched for another framework.

The difference at that line is that, previously, we were in a simulated document structure, I which we could locate an element and "click" on it even if element is hidden, as long as it's in the document structure, like a sub-menu.

But now that we are in a real rendered page, it seems I cannot click on an hidden element. Following code just waits indefinitely until test timeout:

```javascript
    // Go to visualization page
    const menu = await page.getByTitle(
      /Go to experiments visualizations/
    );
    await menu.click();  // waits indefinitely here
```
So, I need to first click on top-menu, to make sub-menu appear, and then click to visible sub-menu.

**Comment by notoraptor** [comment]
I relaunched failing tests (test-long-algo).

**Comment by notoraptor** [comment]
Rebased!

**Pull request merged**